server/src/main/java/org/apache/uniffle/server/buffer/ShuffleBufferManager.java

```diff
@@ -80,11 +84,16 @@ public class ShuffleBufferManager {
   protected Map<String, Map<Integer, AtomicLong>> shuffleSizeMap = JavaUtils.newConcurrentMap();
 
   public ShuffleBufferManager(ShuffleServerConf conf, ShuffleFlushManager shuffleFlushManager) {
+    this.nettyServerEnabled = conf.get(ShuffleServerConf.RPC_SERVER_TYPE) == ServerType.GRPC_NETTY;
```

Comment:
You wanna pass `nettyServerEnabled` from `ShuffleServer` into `ShuffleBufferManager`?

Reply:
Yes

Reply:
This should be fixed.

Reply:
That would add an extra parameter in `ShuffleBufferManager`'s constructor. 

![image](https://github.com/apache/incubator-uniffle/assets/13834479/d1814016-7002-4e04-bb4f-2f66a1cb232e)

It is acceptable? Because the parameter `ShuffleServerConf` has already contained the value of `nettyServerEnabled` indirectly.

Reply:
The single source principle should be ensured. And the checking logic is not same, you can see the current code in shuffleServer.java


```
    nettyServerEnabled = shuffleServerConf.get(ShuffleServerConf.NETTY_SERVER_PORT) >= 0;
    if (nettyServerEnabled) {
      streamServer = new StreamServer(this);
    }
```

Reply:
I'll fix this in the next PR which will fix https://github.com/apache/incubator-uniffle/pull/1008 together(mentioned in https://github.com/apache/incubator-uniffle/pull/1531). If I change this in the current PR, plenty of UTs will be failed. @zuston

Reply:
Ok. let me merge this.

Reply:
> The single source principle should be ensured. And the checking logic is not same, you can see the current code in shuffleServer.java
> 
> ```
>     nettyServerEnabled = shuffleServerConf.get(ShuffleServerConf.NETTY_SERVER_PORT) >= 0;
>     if (nettyServerEnabled) {
>       streamServer = new StreamServer(this);
>     }
> ```

Done in https://github.com/apache/incubator-uniffle/pull/1540.